python/lsst/meas/extensions/trailedSources/NaivePlugin.py

```diff
@@ -292,7 +299,7 @@ def fail(self, measRecord, error=None):
         if error is None:
             self.flagHandler.handleFailure(measRecord)
         else:
-            self.flagHandler.handleFailure(measRecord, error.cpp)
+            self.flagHandler.handleFailure(measRecord.getId(), error)
```

Comment:
This has to be the record, not the id, and I think you do need `.cpp` here. Your comment to me about removing `fail` is worth considering, but I think we should do that on a separate ticket: if `fail` isn't defined and an exception occurs, we'll get a `NotImplementedError` in addition to the traceback. I'm not sure if that's what we actually want here or not, so why don't you file another ticket to look into it.
```suggestion
            self.flagHandler.handleFailure(measRecord, error.cpp)
```